pkg/asset/installconfig/gcp/client.go

```diff
@@ -109,6 +115,28 @@ func (c *Client) GetPublicDNSZone(ctx context.Context, project, baseDomain strin
 	if res == nil {
 		return nil, errors.New("no matching public DNS Zone found")
 	}
+
+	return res, nil
+}
+
+// GetPublicDNSZoneByID returns a public DNS zone for a basedomain by its ID
+func (c *Client) GetPublicDNSZoneByID(ctx context.Context, project, id string) (*dns.ManagedZone, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
+	defer cancel()
+
+	svc, err := c.getDNSService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := svc.ManagedZones.Get(project, id).Context(ctx).Do()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get DNS Zone")
+	}
+	if res.Visibility == "private" {
+		return nil, errors.New(fmt.Sprintf("%s must be public zone, but %s zone provided", id, res.Visibility))
+	}
+
 	return res, nil
 }
```

Comment:
also this seems to be not used outside https://github.com/openshift/installer/pull/2764/commits/73f905029e24a98f1672fac35e5960690a311c6c#r386543216

and should be dropped together with ^^